README.md

```diff
@@ -30,6 +30,9 @@ on:
       - edited
       - synchronize
 
+permissions:
```

Comment:
With this changing to read, can `pull_request_target` above be changed to just `pull_request`?

I ask as a new user I was banging my head trying to figure out why the new workflow didn't trigger. Took longer than I would like to admit to figure out that `pull_request_target` only triggers after it's in base hence it's actually impossible to test during PR submission that it's working as intended.

Reply:
I do not understand why this change affects the `on` part? Can you explain that further to me, please?

I think, @amannn added `pull_request_target` by design here.

Reply:
I think it was added for security given it needed write so someone could put in a pr and use it to write to the repo before it was approved and merged.

With bit limited to read in your update it would be safe to switch to `pull_request` which allows testing the change prior to merge.

Reply:
But actually, you can test changes before merging. Check out my referenced PR in my forked repo for this PR: https://github.com/natterstefan/action-semantic-pull-request/pull/1.

You have to create

- one PR for the origin repo
- one PR (only for demo and testing purposes) in your forked repo

That should do the job without changing this.

Reply:
To clarify I meant testing a PR to add this to your own repo for the first time.

Having to merge a PR before you can confirm it works is counter to best practices so having that be the default in the README definitely caught me out, causing much head scratching.

So if the permissions added mean that it can now be safely used for the standard case with just `pull_request` that would be good.

Reply:
Hi @stevenh,

> Having to merge a PR before you can confirm it works is counter to best practices so having that be the default in the README definitely caught me out, causing much head scratching.

But at the same time, maintainers can be sure that only actions run which were at least added once on the `main` branch, don't they? If so (but I am not 100% sure and not up to date about what is the common way to handle this), I'd leave it as is.

I think this is up to @amannn to decide how to approach your suggestion, but IMHO not in this branch but in another one.

Reply:
I'm definitely open to discuss changing the suggested event trigger, but as @natterstefan said, we can do this in a separate PR to not block this one (as it's already been open for a longer time).